hugolib/author.go

```diff
@@ -13,23 +13,51 @@
 
 package hugolib
 
+import (
+	"fmt"
+	"regexp"
+	"sort"
+	"strings"
+
+	"github.com/spf13/cast"
+)
+
+var (
+	onlyNumbersRegExp = regexp.MustCompile("^[0-9]*$")
+)
+
 // AuthorList is a list of all authors and their metadata.
 type AuthorList map[string]Author
 
+// Authors returns a slice of authors sorted by ID
+func (al AuthorList) Authors() []Author {
```

Comment:
* All the sorting stuff was kind of more and slightly different from what I suggested. I suggest we remove all the different sort methods for now (too many and too much boilerplate), 
* Add `Weight` to Author. This is in line with the reset of Hugo. Less magical than changing the ID from `bep` to `111bep` to make sure I'm the most important.
* Add an `Authors` type to it with the `Less` func etc. that sorts by weight, id, ...? some sensible fallbacks.


The tests can wait until we get the `Multilingual` feature in (will be much easier then).